tangos/input_handlers/halo_stat_files/__init__.py

```diff
@@ -220,4 +226,27 @@ class AmigaIDLStatFile(HaloStatFile):
     def filename(cls, timestep_filename):
         return timestep_filename + '.amiga.stat'
 
-
+    def iter_rows_raw(self, *args):
+        """
+        Yield the halo ID and requested columns from each line of the stat file, without any emulation.
+        If _id_from_file_pos is True, the ID is not taken from the stat file, but rather from the order in which
+        the halos appear.
+        :param args: strings for the column names
+        :return: id, arg1, arg2, arg3 where ID is the halo ID and argN is the value of the Nth named column
+        """
+        with open(self.filename) as f:
```

Comment:
The code expects from the stat file enumerator output of the form `catalog_index, finder_id, arg1, arg2, arg3, ...` but for this particular stat file catalog_index (the number that the halo catalog associates with a given halo) is always equal to the finder_id. That's why there needs to be a different function here.

Reply:
Is the docstring incorrect in that case?  (I think this is also true for the base class method above)

As for avoiding copy-paste, would something like the following work? (NB I didn't actually try it at all! Just speculating)

```python
def iter_rows_raw(self, *args):
   for row in super(self).iter_rows_raw(*args):
       row[0] = row[1] # sequential catalog index not right in this case; overwrite to match finder id
       yield row
```

Reply:
oh yeah I think that's a good solution. I can code that up and see. I'll double check the docstrings

Reply:
I've implemented this and tested (no "self" in the super call, but otherwise exactly as you have it) and it seems to work. I've pushed this over just now!

Reply:
I also double checked and the `iter_rows_raw` docstrings were accurate but `iter_rows` one was not. That has been updated.